Sources/MongoSwift/MongoCollection+Read.swift

```diff
@@ -129,23 +126,8 @@ extension MongoCollection {
             throw MongoError.commandError(message: toErrorString(error))
         }
 
-        let fakeReply: Document = [
-            "ok": 1,
-            "cursor": [
-                "id": 0,
-                "ns": "",
-                "firstBatch": [reply]
-            ] as Document
-        ]
-
-        // mongoc_cursor_new_from_command_reply will bson_destroy the data we pass in,
-        // so copy it to avoid destroying twice (already done in Document deinit)
-        let fakeData = bson_copy(fakeReply.data)
-        guard let newCursor = mongoc_cursor_new_from_command_reply(client._client, fakeData, 0) else {
-            throw MongoError.invalidResponse()
-        }
-
-        return MongoCursor(fromCursor: newCursor, withClient: client)
+        // if no error, reply will always have a values array
+        return reply["values"] as! [BsonValue?]
```

Comment:
why `as!` if the method already throws?

Reply:
it seemed like the only situation in where this wouldn't exist is if there is some change in how the server structures the reply document. otherwise if `mongoc_collection_read_command_with_opts` succeeds, we should always expect this field to be present, and be an array. but if you think we should be more defensive, I can throw an error if the cast fails

Reply:
I think strict errors (e.g. `as!`) works well for cases where we want to trust libmongoc's API, but anything involving a server response should be more defensive and fail with exceptions/errors instead of hard assertions.